airflow/models/taskmixin.py

```diff
@@ -69,6 +70,14 @@ def set_downstream(
         """Set a task or a task list to be directly downstream from the current task."""
         raise NotImplementedError()
 
+    def as_setup(self):
```

Comment:
Should we mark this and the other one as an `abstractmethod`?

Reply:
reason i didn't do this is i don't necessarily want all subclasses to implement.  e.g. it's not supported for mapped operator, which is a different xcomarg class from PlainXComArg cc @uranusjr if you have thoughts on this PR